packages/core/contracts/common/implementation/AncillaryData.sol

```diff
@@ -54,6 +80,16 @@ library AncillaryData {
         } else {
             prefix = abi.encodePacked(key, ":");
         }
-        return abi.encodePacked(currentAncillaryData, prefix, AncillaryData.toUtf8Bytes(value));
+        return abi.encodePacked(currentAncillaryData, prefix, toUtf8Bytes(value));
+    }
+
+    function appendChainId(bytes memory currentAncillaryData) internal view returns (bytes memory) {
```

Comment:
Have you thought about just having a single overloaded method:
```
function appendKeyValue(
        bytes memory currentAncillaryData,
        bytes memory key,
        address value
    );

function appendKeyValue(bytes memory currentAncillaryData, bytes memory key, uint256 value) internal view returns (bytes memory);
```

And then the caller can choose to put pass in the chain id if they like... thoughts?

Reply:
I like that

Reply:
I ended up refactoring the methods to use a common `internal` method because I found that after overloading the methods, the `PerpetualLib` contract ended up being over the bytecode limit. Not sure what happened here, but PTAL at this [PR](https://github.com/UMAprotocol/protocol/pull/3054/commits/1f4c72d24de1d51a5a331f79ae6cc28bf7a48397)